### ISSUE TITLE
Use of MenuView without PagingMenuController

### ISSUE THREAD

**Issue by LiborZ** [kitasuke/PagingMenuController#153]
Hi,

is it possible to use MenuView class (and MenuItemView and PagingMenuOptions) without PagingMenuController? I like your paging menu control but I want to use it little different way. I don't want to create controller for each item because I want to use just one controller and change data in it. So it it possible? I am trying this but for now without luck:

```
let menuItemTitles = ["Yesterday", "Today", "Tomorrow"]
let options = PagingMenuOptions()
options.menuItemCount = 3
options.menuDisplayMode = PagingMenuOptions.MenuDisplayMode.Standard(widthMode: .Flexible, centerItem: true, scrollingMode: .ScrollEnabled)
options.menuHeight = 60
    
let menuView = MenuView(menuItemTitles: menuItemTitles, options: options)
menuView.backgroundColor = UIColor.grayColor()
self.view.addSubview(menuView)
```

Nothing is shown in view controller. I tried add and call:

```
private func layoutMenuView(menuView: MenuView, options: PagingMenuOptions) {
    let viewsDictionary = ["menuView": menuView]
    let metrics = ["height": options.menuHeight]
    let horizontalConstraints = NSLayoutConstraint.constraintsWithVisualFormat("H:|[menuView]|", options: [], metrics: nil, views: viewsDictionary)
    let verticalConstraints: [NSLayoutConstraint]
    switch options.menuPosition {
    case .Top:
      verticalConstraints = NSLayoutConstraint.constraintsWithVisualFormat("V:|[menuView(height)]", options: [], metrics: metrics, views: viewsDictionary)
    case .Bottom:
      verticalConstraints = NSLayoutConstraint.constraintsWithVisualFormat("V:[menuView(height)]|", options: [], metrics: metrics, views: viewsDictionary)
    }
    
    NSLayoutConstraint.activateConstraints(horizontalConstraints + verticalConstraints)
    
    menuView.setNeedsLayout()
    menuView.layoutIfNeeded()
  }
```

But still nothing is shown. Thanks for help

**Comment by kitasuke** [comment]
There is no plan to provide MenuView alone for now, but might be good to have one controller usage. It could be suitable for some cases.

**Comment by mitchellporter** [comment]
I've thought about this as well. For example, making the menu view a header for a UITableView or UICollectionView, and then whenever you swipe to the next section, delegate method gets called, and you react by updating the menu.

**Comment by kitasuke** [comment]
I've never thought about making it a header view, but it's a good idea.

**Comment by MarcelStructr** [comment]
I also want to use it exactly like @LiborZ . Is there a good solution now?

**Comment by kitasuke** [comment]
Not for now, but it's one of the highest priority task.

**Comment by MarcelStructr** [comment]
Can't wait for it (:

**Comment by MarcelStructr** [comment]
Hello @kitasuke ,
could you guess how long you will take for implementing this special request? I am currently working on the iOS-Version of http://borntobeastar.com and would gladly add your work into the app and make a reference to it.

**Comment by kitasuke** [comment]
I've just started working on this issue which means top priority task. I guess its less than a couple of weeks!

**Comment by kitasuke** [comment]
`MenuControllerSet` option's added with this PR in order to use only one menu controller. #195 
Please set `Single`.

This PR allows you to use MenuView and MenuController separately with `ComponentType` option. #197 
Can you check PagingMenuControllerDemo2 in example project to see how it works?

**Comment by MarcelStructr** [comment]
Hello @kitasuke , thank you for your fast work!!!
I don't exactly get, how to use it properly right now. In the example you still work with Controllers, but I don't want to include Controller in the code, because everything happens in one Controller and I want to create the View there. 

I hoped to use it like that, but I don't see anything on the display. Can you help me out?

```
let options = PagingMenuOptions()
                options.menuItemMargin = 4
                options.menuHeight = 60
                options.menuDisplayMode = .SegmentedControl
                options.menuComponentType = .Separated
                options.menuControllerSet = .Single
               
                
let menuView = MenuView(menuItemTitles: ["Category1", "Category2", "Category3", "Category4"], options: options)
                
 self.view.addSubview(menuView)
```

**Comment by kitasuke** [comment]
Got it, I'll fix it. It's much simpler to use.

**Comment by MarcelStructr** [comment]
Hello @kitasuke , did you already fix it?

**Comment by kitasuke** [comment]
I'll restart fixing it from tonight. The progress is half done.

**Comment by kitasuke** [comment]
It's done!
Please check usage with example project in master branch.

**Comment by MarcelStructr** [comment]
Thank you! I will check it out tomorrow and leave a reply.

**Comment by kitasuke** [comment]
New version, 0.10.0 has released. It includes the feature above, so please check it out.

**Comment by MarcelStructr** [comment]
I checked your new examples out, and they are really amazing. The Roundrect-Option is what i was looking for and in the example project it works well.
Now I tried to extract the important code snippets into a separate project. But again I get this weird error:
http://www2.pic-upload.de/img/30686665/error.jpg

Below is my source code, what did I miss? I included the new 0.10.0 version. I tried both ways by Storyboard and by coding only, same error. In your example project it works, though.
http://www2.pic-upload.de/img/30706873/paging.jpg

**Comment by MarcelStructr** [comment]
Also is it possible to put two MenuViews in one Controller like this? 
http://www2.pic-upload.de/img/30706916/UIScrollView.jpg

I hope I didn't show you this too late. :/

**Comment by kitasuke** [comment]
Can you delete your project folder in derivedData and try again? The path is here `/Users/Library/Developer/Xcode/DerivedData/`.

If you want to simply put two MenuViews, you can use two PagingMenuController which have MenuView component type and All component type, but it depends on your expected behaviour.
Or do you want to show menu item view vertically instead of horizontally which is current way?

**Comment by MarcelStructr** [comment]
The MenuView below is dependent on which item got chosen on the MenuView above.
So for example: http://www2.pic-upload.de/img/30706916/UIScrollView.jpg 
If I choose "Sports" at he MenuView above, certain items will be loaded on the MenuView under it ("BMX", "Acrobatics", and more).

Is this behaviour possible? I just want to show it horizontally, like on the picture.

I will be on my Mac in a few hours, and will delete the project folder in derivedData and leave another reply. Thank you again for your help, you are a great person.

**Comment by kitasuke** [comment]
I think I get it.
Well, it's not possible with one paging menu controller, but could be possible with two paging menu controller below.

```
let topOptions = PagingMenuOptions()
topOptions.menuComponentType = .MenuView
let topPagingMenuController = PagingMenuController(menuItemTypes: ["Music", "Sports", "Entertainment"], options: topOptions)

let viewController1 = UIViewController()
viewController1.title = "Acrobatics"
let viewController2 = UIViewController()
viewController2.title = "BMX"
let viewController3 = UIViewController()
viewController3.title = "Parkour"

let bottomOptions = PagingMenuOptions()
let bottomPagingMenuController = PagingMenuController(menuControllerTypes: [viewController1, viewController2, viewController3], options: topOptions)
```

However, this library isn't designed for this behaviour. So this usage could need so much code than you expected or simply take memory usage.

**Comment by MarcelStructr** [comment]
If it works, I will be totally fine. Thanks for the code. However at the moment I can't test the code snippet, because I still get the error. 
I deleted the folder in DerivedData like you said and also tried out a new project, where I downloaded the library (version  (0.10.1)) via Cocoapods. I also tested the code from PagingMenuControllerDemo2 in my separate project and also same error. It seems to be a general issue, or I am missing something.

**Comment by kitasuke** [comment]
Did you `pod install` after deleting the folder?

**Comment by MarcelStructr** [comment]
Okay I found the issue. If you want to check out the issue for yourself, create a simple Single View Application and use the ViewController, that was generated with the project. If you use the library in that ViewController it will crash. After I added a Navigation Controller and other Views it suddenly started to work.

**Comment by MarcelStructr** [comment]
I am also nearly done with what I want to achieve:

I got two MenuViews in a row by creating two Container Views:
http://www2.pic-upload.de/img/30723286/page2.jpg

This is the source code. I simply created two different MenuViews and enabled one to each Container View:
http://www2.pic-upload.de/img/30723283/page1.jpg

The only problem now: The delegate methods don't work. This is the last part, where I need your help, then it's done!
